src/script_opt/ZAM/BuiltInSupport.cc

```diff
@@ -68,10 +68,24 @@ void FixedCatArg::RenderInto(ZVal* zframe, int slot, char*& res) {
             break;
 
         case TYPE_DOUBLE:
-        case TYPE_TIME:
-            n = modp_dtoa2(z.AsDouble(), res, 6);
+        case TYPE_TIME: {
+            auto d = z.AsDouble();
+            n = modp_dtoa2(d, res, 6);
             res += n;
+
+            // Taken from ODesc::Add(double d, bool no_exp)
+            auto approx_equal = [](double a, double b, double tolerance = 1e-6) -> bool {
```

Comment:
We should move this to util, since I can see it being useful in other places.

Reply:
Okay, I'll update with that change.

Reply:
@timwoj fixup now pushed